java-surefire/src/main/java/org/sonar/plugins/surefire/api/SurefireUtils.java

```diff
@@ -28,46 +28,95 @@
 
   private static final Logger LOGGER = Loggers.get(SurefireUtils.class);
+  /**
+   * @deprecated since 4.11
+   */
+  @Deprecated
   public static final String SUREFIRE_REPORTS_PATH_PROPERTY = "sonar.junit.reportsPath";
+  /**
+   * @since 4.11
+   */
+  public static final String SUREFIRE_REPORT_PATHS_PROPERTY = "sonar.junit.reportPaths";
 
   private SurefireUtils() {
   }
 
   /**
-   * Find the directory containing the surefire reports.
+   * Find the directories containing the surefire reports.
    * @param settings Analysis settings.
    * @param fs FileSystem containing indexed files.
    * @param pathResolver Path solver.
-   * @return The directory containing the surefire reports or default one (target/surefire-reports) if not found (not configured or not found).
+   * @return The directories containing the surefire reports or default one (target/surefire-reports) if not found (not configured or not found).
    */
-  public static File getReportsDirectory(Settings settings, FileSystem fs, PathResolver pathResolver) {
-    File dir = getReportsDirectoryFromProperty(settings, fs, pathResolver);
-    if (dir == null) {
-      dir = new File(fs.baseDir(), "target/surefire-reports");
+  public static List<File> getReportsDirectories(Settings settings, FileSystem fs, PathResolver pathResolver) {
+    File dir = getReportsDirectoryFromDeprecatedProperty(settings, fs, pathResolver);
+    List<File> dirs = getReportsDirectoriesFromProperty(settings, fs, pathResolver);
+    if (dirs != null) {
+      if (dir != null) {
+        // both properties are set, deprecated property ignored
+        LOGGER.info(String.format("Property '%s' is deprecated and will be ignored, as property '%s' is also set.",
+          SUREFIRE_REPORTS_PATH_PROPERTY,
+          SUREFIRE_REPORT_PATHS_PROPERTY));
+      }
+      return dirs;
     }
-    return dir;
+    if (dir != null) {
+      return Collections.singletonList(dir);
+    }
+    // both properties are not set
+    return Collections.singletonList(new File(fs.baseDir(), "target/surefire-reports"));
   }
 
   @CheckForNull
-  private static File getReportsDirectoryFromProperty(Settings settings, FileSystem fs, PathResolver pathResolver) {
+  private static List<File> getReportsDirectoriesFromProperty(Settings settings, FileSystem fs, PathResolver pathResolver) {
+    if(settings.hasKey(SUREFIRE_REPORT_PATHS_PROPERTY)) {
+      String paths = settings.getString(SUREFIRE_REPORT_PATHS_PROPERTY);
+      if (paths != null) {
+        return Arrays.stream(paths.split(","))
+          .map(String::trim)
+          .map(path -> getFileFromPath(fs, pathResolver, path))
+          .filter(Objects::nonNull)
+          .collect(Collectors.toList());
+      }
+    }
+    return null;
+  }
+
+  @CheckForNull
+  private static File getReportsDirectoryFromDeprecatedProperty(Settings settings, FileSystem fs, PathResolver pathResolver) {
     if(settings.hasKey(SUREFIRE_REPORTS_PATH_PROPERTY)) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "SUREFIRE_REPORTS_PATH_PROPERTY"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)